hardware_interface/include/hardware_interface/handle.hpp

```diff
@@ -24,84 +25,135 @@ namespace hardware_interface
 {
 /** A handle used to get and set a value on a given interface. */
 template<class HandleType>
-class Handle
+class ReadOnlyHandle
 {
 public:
-  HARDWARE_INTERFACE_PUBLIC
-  Handle(
-    const std::string & name, const std::string & interface_name,
+  ReadOnlyHandle(
+    std::string name,
+    std::string interface_name,
     double * value_ptr = nullptr)
-  : name_(name), interface_name_(interface_name), value_ptr_(value_ptr)
+  : name_(std::move(name)), interface_name_(std::move(interface_name)), value_ptr_(value_ptr)
```

Comment:
This feels highly redundant. You changed a pass-by-const-ref to pass-by-value + std::move. I'd really prefer if we didn't abuse `std::move` for petty stuff like this.

Reply:
I think that's a really valid case especially given that these names are mainly short strings, so SSO kicks in perfectly.

Reply:
This is superstitious programming and premature optimization. I trust to compiler to know where to optimize and I am still strongly for keeping the code simple. 

To back up my claim, I've ran google benchmark on this. The code can be found [here](https://gist.github.com/bmagyar/d3771595e7567cb00479a76b3a711cdb). Took the long names from our urdf examples page.
These benchmarks show that the `std::move` alternative (marked with `/0`) has a few nanoseconds of overhead compared to the other one both in debug and release mode. Even if it was the opposite way, I'd argue to keep the code simpler and swallow that 3ns overhead per handle and *avoid std::move*.

```
***WARNING*** Library was built as DEBUG. Timings may be affected.
-----------------------------------------------------------------------------------------
Benchmark                                              Time            CPU       Iterations
------------------------------------------------------------------------------------------
MyFixture<StateHandle>/ShortNames_StateHandle/0       69.0 ns         69.0 ns     10294137
MyFixture<StateHandle>/ShortNames_StateHandle/1       53.5 ns         53.5 ns     12842977
MyFixture<StateHandle>/LongNames_StateHandle/0        87.1 ns         87.1 ns      8088183
MyFixture<StateHandle>/LongNames_StateHandle/1        76.3 ns         76.3 ns      9228388
```
The same in a release build:
```
------------------------------------------------------------------------------------------
Benchmark                                              Time           CPU      Iterations
------------------------------------------------------------------------------------------
MyFixture<StateHandle>/ShortNames_StateHandle/0       17.3 ns         17.3 ns     40833624
MyFixture<StateHandle>/ShortNames_StateHandle/1       13.7 ns         13.7 ns     51205075
MyFixture<StateHandle>/LongNames_StateHandle/0        30.9 ns         30.9 ns     21597366
MyFixture<StateHandle>/LongNames_StateHandle/1        27.5 ns         27.5 ns     25822783
```

((( I made the fixture templated on the handle type but realized that there was no point in testing multiple, sorry for the added complexity)))

Reply:
I think that's a real common use-case. https://www.bfilipek.com/2018/08/init-string-member.html
It doesn't make a big difference when dealing with SSO, you're right.

Reply:
Please get rid of `std::move`. This constructor is now also inconsistent with `ReadWriteHandle`.

Reply:
done in a36352e